src/ServiceControlInstaller.Engine/Instances/ServiceControlInstance.cs

```diff
@@ -147,7 +147,12 @@ protected override void ApplySettingsChanges(KeyValueConfigurationCollection set
             settings.Set(ServiceControlSettings.LogPath, LogPath);
             settings.Set(ServiceControlSettings.ForwardAuditMessages, ForwardAuditMessages.ToString(), Version);
             settings.Set(ServiceControlSettings.ForwardErrorMessages, ForwardErrorMessages.ToString(), Version);
-            settings.Set(ServiceControlSettings.AuditRetentionPeriod, TimeSpan.FromHours(1).ToString(), Version); //TODO: Remove in future
+
+            if (AuditRetentionPeriod.HasValue)
+            {
+                settings.Set(ServiceControlSettings.AuditRetentionPeriod, AuditRetentionPeriod.Value.ToString(), Version);    
```

Comment:
`settings.Set` is how we remove config values that aren't there. Maybe add an `else` that calls `RemoveIfRetired`?

Reply:
Would

```
settings.Set(ServiceControlSettings.AuditRetentionPeriod, AuditRetentionPeriod.ToString(), Version); 
```

work as well?

Reply:
Yes it would

Reply:
Apparently, we have not retired the setting because existing instances that got converted still need it. Maybe we should retire the setting then?

Reply:
Ah. Now that you mention it, we can't actually retire the setting. Existing instances can still have Audit messages in them after the upgrade so we need AuditRetentionPeriod to stay. 

I think `settings.Set` is the best option because it will remove the setting and then only re-add it if it is there